src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -278,14 +295,21 @@ trait ScalaSettings extends AbsScalaSettings
   def optBoxUnbox                = optEnabled(optChoices.boxUnbox)
   def optNullnessTracking        = optEnabled(optChoices.nullnessTracking)
   def optClosureInvocations      = optEnabled(optChoices.closureInvocations)
+  def optInlinerEnabled          = optEnabled(optChoices.inline)
 
-  def optInlineProject           = optEnabled(optChoices.inlineProject)
-  def optInlineGlobal            = optEnabled(optChoices.inlineGlobal)
-  def optInlinerEnabled          = optInlineProject || optInlineGlobal
+  // deprecated inliner levels
+  def optLProject                = optEnabled(optChoices.lProject)
+  def optLClasspath              = optEnabled(optChoices.lClasspath)
 
   def optBuildCallGraph          = optInlinerEnabled || optClosureInvocations
   def optAddToBytecodeRepository = optBuildCallGraph || optInlinerEnabled || optClosureInvocations
 
+  val optInlineFrom = StringSetting(
+    "-opt-inline-from",
+    "patterns",
+    "Classfile name patterns from which to allow inlining. ** = anything, * = package or class name, ! to exclude. Example: scala.**:!scala.Predef$:corp.*.util.*:corp.**.*Util*",
+    "")
```

Comment:
Perhaps this could be a `MultiStringSetting` to allow multiple `-opt-inline-from` entries on the command line to aggregate.

`scaladoc -implicits-hide` is an example of a similar part of the settings UI that goes that way.